### PR TITLE
Ci 13616 upload artifacts to j frog

### PR DESCRIPTION
**Pull request by lkilpatrick-harness**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

https://harness.atlassian.net/browse/CI-13616

Added stage variable examples to https://developer.harness.io/docs/continuous-integration/use-ci/build-and-upload-artifacts/upload-artifacts/upload-artifacts-to-jfrog/

and

https://developer.harness.io/docs/continuous-integration/continuous-integration-faqs/#can-i-run-the-upload-artifacts-to-jfrog-artifactory-step-with-a-non-root-user

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: kb/continuous-integration/continuous-integration-faqs.md [hunk 1]
```diff
@@ -1457,15 +1457,33 @@ If you want to include the volume content in the image layers, consider using `C
...
-Go to the [Kaniko container runtime error article](./articles/kaniko_container_runtime_error).
+Go to the [Kaniko container runtime error article](/kb/continuous-integration/articles/kaniko_container_runtime_error).
+
+### When using 'Build ans Push' steps with Base Image connector, Can I pull and push from two different docker registries that have same prefix for registry URL ?
+
+No, when using base image connector, ensure the prefix of the url use for pulling is different than the prefix of the url in the connector used for pushing. 
+
+If two registry URLs begin with same prefix(e.g. https://index.docker.io) it will result in the second registry credentials getting over-ridden in the docker config file when a docker login is attempted 
+
+In other cases with a docker compatible jfrog registry, this limitation is not present.
```
Comment (nofarblue) [inline kb/continuous-integration/continuous-integration-faqs.md:1476]: @lkilpatrick-harness  the limitation applies to all build and push steps, other than for 'build and push to docker registry " when configured with jfrog?
if so, why do we add this in the FAQ of 'Build and Push to JFrog Docker registries' page ?

**Review by nofarblue — DISMISSED** [review]
hey, i've made a few changes directly to the source branch, also left a comment

**Comment by dewan-ahmed** [comment]
Hi @lkilpatrick-harness pls address Nofar's comment(s) and the MC. thanks

File: docs/continuous-integration/use-ci/build-and-upload-artifacts/build-and-push/build-and-push-to-docker-jfrog.md [hunk 2]
```diff
@@ -209,3 +209,4 @@ Go to the [CI Knowledge Base](/kb/continuous-integration/continuous-integration-
 * [Why doesn't the Build and Push step include the content of VOLUMES from my Dockerfile in the final image?](/kb/continuous-integration/continuous-integration-faqs/#why-doesnt-the-build-and-push-step-include-the-content-of-volumes-from-my-dockerfile-in-the-final-image)
 * [Can I use a specific version of kaniko or drone-docker?](/kb/continuous-integration/continuous-integration-faqs/#is-there-a-way-to-use-a-newer-or-older-version-of-kaniko)
 * [How do I fix this kaniko container runtime error: kaniko should only be run inside of a container?](/kb/continuous-integration/articles/kaniko_container_runtime_error)
+* [Can I push and pull from two different docker registries that have same prefix for registry URL ?](/kb/continuous-integration/continuous-integration-faqs/#can-i-push-and-pull-from-two-different-docker-registries-that-have-same-prefix-for-registry-url-)
```
Comment (nofarb) [inline docs/continuous-integration/use-ci/build-and-upload-artifacts/build-and-push/build-and-push-to-docker-jfrog.md:212]: another comment said this limitation doesn't apply in jfrog. should this be removed? @lkilpatrick-harness

Comment (Aishwarya-Lad) [reply docs/continuous-integration/use-ci/build-and-upload-artifacts/build-and-push/build-and-push-to-docker-jfrog.md:212]: yea we can remove this from here, and also from FAQs of all the build and push steps(ACR, GCR, GHCR, ECR, GAR) except of docker.

**Review by dewan-ahmed — APPROVED** [review]
LGTM.

**Pull request merged**